### PR TITLE
iox-1391 move functional files

### PR DESCRIPTION
**Pull request by FerdinandSpitzschnueffler**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

See PR title.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1391 (partially)

### CODE REVIEW

**Review by elBoberido — DISMISSED** [review]

File: doc/website/release-notes/iceoryx-unreleased.md [hunk 1]
```diff
@@ -605,20 +605,20 @@
...
 
-27. `cxx::function` is no longer nullable.
+27. `iox::function` is no longer nullable.
 
     ```cpp
     // before
-    cxx::function<void()> helloFunc = []{ std::cout << "hello world\n"; };
-    cxx::function<void()> emptyFunction;
+    iox::function<void()> helloFunc = []{ std::cout << "hello world\n"; };
+    iox::function<void()> emptyFunction;
```
Comment (mossmaurice) [inline doc/website/release-notes/iceoryx-unreleased.md:613]: This should not be changed:

```suggestion
    iox::cxx::function<void()> helloFunc = []{ std::cout << "hello world\n"; };
    iox::cxx::function<void()> emptyFunction;
```

File: iceoryx_hoofs/legacy/include/iceoryx_hoofs/internal/cxx/storable_function.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
```
Comment (mossmaurice) [inline iceoryx_hoofs/legacy/include/iceoryx_hoofs/internal/cxx/storable_function.hpp:1]: @FerdinandSpitzschnueffler I think you can delete this file here. As this is internal API, there is no need to provide a legacy header.

File: iceoryx_posh/include/iceoryx_posh/popo/wait_set.hpp [hunk 3]
```diff
@@ -30,6 +28,8 @@
 #include "iceoryx_posh/popo/trigger_handle.hpp"
 #include "iceoryx_posh/runtime/posh_runtime.hpp"
 #include "iox/algorithm.hpp"
+#include "iox/function.hpp"
```
Comment (mossmaurice) [inline iceoryx_posh/include/iceoryx_posh/popo/wait_set.hpp:31]: Can't this include be removed?

**Review by mossmaurice — COMMENTED** [review]
LGTM in general, few minor findings.

File: iceoryx_posh/include/iceoryx_posh/popo/wait_set.hpp (hunk 3 above)
Comment (FerdinandSpitzschnueffler) [reply iceoryx_posh/include/iceoryx_posh/popo/wait_set.hpp:31]: No, a `function` is used in `attachImpl`.

**Review by mossmaurice — APPROVED** [review]

**Review by elBoberido — APPROVED** [review]

**Pull request merged**